rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -62,6 +69,42 @@ SequentialCompressionWriter::~SequentialCompressionWriter()
 
 void SequentialCompressionWriter::compression_thread_fn()
 {
+  if (compression_options_.thread_nice_value) {
+
+#ifdef _WIN32
+    ROSBAG2_COMPRESSION_LOG_WARN_STREAM("Lowering of process priority is not implemented for windows");
+
+    /**
+     * The implementation should look something like this
+
+    uint8_t nice_value = *compression_options_.thread_nice_value;
+
+    // this must match THREAD_PRIORITY_IDLE, THREAD_PRIORITY_LOWEST...
+    DWORD dwThreadPri = *compression_options_.thread_nice_value;
+
+    if(!SetThreadPriority(GetCurrentThread(), dwThreadPri))
+    {
+      ROSBAG2_COMPRESSION_LOG_WARN_STREAM(
+        "Could not set nice value of compression thread to " << nice_value << " : " << std::strerror(
+          errno));
+    }
+    */
+
+#else
+    int wanted_nice_value = *compression_options_.thread_nice_value;
```

Comment:
If I would do that, the error output down below would be broken, as it would interpret wanted_nice_value as char, or I would need to use static cast all over the place.